engine-nextgen/src/main/kotlin/graphql/nadel/validation/NadelTypeValidation.kt

```diff
@@ -98,6 +106,31 @@ internal class NadelTypeValidation(
         return typeErrors + listOfNotNull(outputTypeError)
     }
 
+    /**
+     * Answers whether `rhs` assignable to `lhs`?
+     *
+     * i.e. does the following compile
+     *
+     * ```
+     * vol output: lhs = rhs
+     * ```
+     *
+     * Note: this assumes both types are from the same schema. This does NOT
+     * deal with differences between overall and underlying schema.
+     */
+    fun isAssignableTo(lhs: GraphQLNamedOutputType, rhs: GraphQLNamedOutputType): Boolean {
+        if (lhs.name == rhs.name) {
+            return true
+        }
+        if (lhs.name == GraphQLID.name && rhs.name == GraphQLString.name) {
```

Comment:
I'm going to leave this in… I think the other type validation should follow this in the future? Thoughts?

cc: @felipe-gdr

Reply:
so this allows String -> ID, and one way only?

Reply:
Yep. I think the other way around is technically possible too. They're both Strings, but as an input type ID accepts Ints too.